posts/solving-parallel-letter-frequency-elixir-part-3-benchmarking-code.md

```diff
@@ -1,21 +1,29 @@
----
-title: "Solving Parallel Letter Frequency in Elixir - Part III: Benchmarking the Code"
-date: 2018-06-27T21:29:19+08:00
-draft: true
----
-In `Part I` I showed how I applied Unicode matching in Elixir to help me solve the first part of [Parallel Letter Frequency](https://exercism.io/tracks/elixir/exercises/parallel-letter-frequency), a medium difficulty problem on [Exercism's Elixir Track](https://exercism.io/tracks/elixir). In `Part II` I tackled the next part of the problem -- adding concurrency -- and explored how you can use Elixir's `Task` module to do this.
+# 3. What Parallel Letter Frequency can teach you about benchmarking in Elixir
 
-In this final part I'll walk you through how to benchmark the Elixir code to determine whether or not adding concurrency actually made it any faster. As you'll see, although concurrency might be seen as a "free" way to make your code faster, in some cases it can have the opposite effect. 
+Exercises on Exercism are small, synthetic, and often seemingly trivial. It’s easy to imagine that experienced practitioners would have nothing to learn from them. However, solving these synthetic problems can push you to learn and apply parts of your language that you may not have explored. This new learning can lead you to solve real world problems more efficiently or in a more expressive way.
 
-The concurrent code in my solution ran around **80% faster** on my system for certain workloads, but in some cases was over **3x slower than the original code**. As the benchmarks later in the post demonstrate, depending on the situation, adding concurrency can actually **reduce performance while also increasing the complexity of your code**.
+[Parallel Letter Frequency](https://exercism.io/tracks/elixir/exercises/parallel-letter-frequency) is a medium difficulty exercise on [Exercism's Elixir Track](https://exercism.io/tracks/elixir). It asks you to write a function that calculates the frequency of letters in a list of strings, and to do so concurrently. This unpacks a surprising number of interesting lessons.
```

Comment:
Here, too, we need a transition from the intro that introduces what this particular blog post is about.